### DISCUSSION TITLE
Can I use `useMutation` with GET request?

### DISCUSSION THREAD

**Discussion by athrunsun** [TanStack/query#3675] [Q&A] [answered]
Searching online, I found [this](https://stackoverflow.com/questions/62340697) and I know I can use `enabled` option to defer the request sending.

But someone is asking [here](https://stackoverflow.com/questions/62340697/react-query-how-to-usequery-when-button-is-clicked?rq=1#comment118616741_63113066),

> Why not using `useMutation` instead of `useQuery`?

I have the same question, but I'm not sure whether `useMutation` is designed for a GET request.

The [official documentation](https://react-query.tanstack.com/guides/mutations) doesn't state explicitly that it cannot be used with a GET request,

> mutations are typically used to create/update/delete data

So, can I use `useMutation` with GET request?

**Comment by TkDodo** [comment, marked as answer]
sure you can, but difference are that state of mutations is not shared across instances, and there is no automatic / smart refetching.

**Reply by julianadeveloper** [reply]
Please, can you exemplify this in a use case?
example:
I have an API with a POST request. I send an object containing an ID in the body and the API returns a response containing a result.
In my frontend, I have a service that uses the POST method for this endpoint and in the hook it contains a query that I use useQuery, but I tried using useMutation and I had no success at all. Not even using promise.

**Reply by TkDodo** [reply]
you can use queries with post request, as long as they don’t have any side-effects on the server. GET and POST are rest verbs that imply intent: GET is an idempotent request that can be cached by the browser and can be sent as often as you like. POST has a body and is meant for updates.

If you use POST to fetch some data and you don’t care if this runs more than once, you can use it in a useQuery (react-query really doesn’t care - in graphQL, everything is a POST). If you use a query to send an email to someone with a POST request, this wouldn’t be good, that’s what mutations are for.

**Comment by muhammad-paksi** [comment]
@TkDodo, Can we please ask for a separate function for get operations, say useFetching or useAction, to avoid confusion about whether useMutation can only be used to mutate data?
